ci: fix symbol test to handle duplicate symbols

kara opened the pull request:
In 7cb8396, we improved the symbol test to remove suffixes from
the output, so "CIRCULAR$1" became "CIRCULAR". However, the logic
that checked for extra symbols depended on each symbol being unique.
If multiple symbols with the same name were added (e.g. pulled in
from separate files), they would be added to the map as "extras",
even if they were marked as expected in the golden file.

This commit updates the symbol checking logic to take multiple
symbols with the same name into account.

Closes #28406

alexeagle commented on tools/symbol-extractor/symbol_extractor.ts:
```diff
@@ -12,6 +12,11 @@ import * as ts from 'typescript';
 
+interface SymbolData {
+  type: 'missing'|'extra';
+  count: number;
```
opt: maybe a docstring about how the count is used? it's only for printing to the user, right? and doesn't change the content of the golden file?

kara committed:
ci: fix symbol test to handle duplicate symbols

In 7cb8396, we improved the symbol test to remove suffixes from
the output, so "CIRCULAR$1" became "CIRCULAR". However, the logic
that checked for extra symbols depended on each symbol being unique.
If multiple symbols with the same name were added (e.g. pulled in
from separate files), they would be added to the map as "extras",
even if they were marked as expected in the golden file.

This commit updates the symbol checking logic to take multiple
symbols with the same name into account.

Closes #28406

kara committed:
fixup! ci: fix symbol test to handle duplicate symbols

alexeagle reviewed (approved):
nice!

kara commented:
merge-assistance: integration test failing due to https://github.com/angular/angular/pull/28460
